vagrant/README.md

```diff
@@ -0,0 +1,79 @@
+# About
+
+This is a fairly minimal Vagrantfile and associated bootstrap scripts for
+setting up a build environment for
+[TrustedFirmware-M](https://www.trustedfirmware.org).
+
+Note that the vagrant virtual machine will contain an independent copy of TF-M
+and the mbed-os-tf-m-regression-tests.
+
+Note that by default vagrant shares `/vagrant` in the virtual machine with the
+host. You can use this directory to copy binaries from inside the vagrant
+machine out to the host for e.g. programming via USB.
+
+# Howto
+
+### TF-M + Mbed OS Regression Tests
+
+The following quickstart instructions will build you the TF-M regression tests
+for MUSCA_B1 on Mbed OS.
+
+```
+$ cd vagrant
+$ vagrant up
+$ vagrant ssh
+$ cd mbed-os-tf-m-regression-tests
+$ mbed deploy
+$ python3 build_tfm.py -m CY8CKIT_064S2_4343W -t GNUARM -c ConfigRegressionIPC.cmake
+```
+
+### TF-M Standalone Regression Tests
+
+These quickstart instructions will build you the TF-M regression tests for
+MUSCA_B1 standalone (without Mbed OS).
+
+```
+$ cd vagrant
+$ vagrant up
+$ vagrant ssh
+$ cd tfm/trusted-firmware-m
+$ mkdir cmake_build
+$ cd cmake_build
+$ cmake -G"Unix Makefiles" -DPROJ_CONFIG=`readlink -f ../configs/ConfigRegressionIPC.cmake) -DTARGET_PLATFORM=MUSCA_B1 -DCOMPILER=GNUARM -DCMAKE_BUILD_TYPE=Release ../
```

Comment:
cmake -G"Unix Makefiles" -DPROJ_CONFIG=`readlink -f ../configs/ConfigRegressionIPC.cmake` -DTARGET_PLATFORM=MUSCA_B1 -DCOMPILER=GNUARM -DCMAKE_BUILD_TYPE=Release ../
No bracket - -DPROJ_CONFIG=`readlink -f ../configs/ConfigRegressionIPC.cmake)

Reply:
Getting - cmake config error, if the above cmake command is corrected - following the instructions
```-- embedded_project_start: using project specific config file (PROJ_CONFIG = /home/vagrant/tfm/trusted-firmware-m/configs/ConfigRegressionIPC.cmake)
-- Using gcc compiler package v9.2.1 from /home/vagrant/.local
-- embedded_project_start: using project specific config file (PROJ_CONFIG = /home/vagrant/tfm/trusted-firmware-m/configs/ConfigRegressionIPC.cmake)
-- Using gcc compiler package v9.2.1 from /home/vagrant/.local
CMake Error at app/CMakeLists.txt:36 (message):
  Missing CMSIS.  Please clone the tf-m-tests repo.
```

Reply:
Interesting. tf-m-tests is new, isn't it? I think this means our `master` is now broken as well, when trying to build TF-M's master branch. This isn't an issue caused by this PR, so no action needed in this PR. We could specify which tag of TF-M to use though when cloning it: use a version old enough that we don't need the tf-m-tests repo.

Reply:
We use - https://github.com/ARMmbed/trusted-firmware-m/commits/dev/feature-dualcore for this repo. We have not switched to TF-M https://git.trustedfirmware.org/TF-M/trusted-firmware-m.git/log/ yet. So probably we need to update example instructions, as this is TF-M only - https://git.trustedfirmware.org/TF-M/trusted-firmware-m.git/tree/docs/getting_started/tfm_build_instruction.rst

Reply:
Updated vagrant to clone the extra repos needed, even those not mentioned in tfm_build_instruction.rst yet.

Reply:
Updated this so that one can follow the instructions in the README without running into error. Rather than force use of an older TF-M, I added the new repos to clone.